homeassistant/components/vacuum/services.yaml

```diff
@@ -131,3 +131,19 @@ xiaomi_remote_control_move_step:
     duration:
       description: Duration of the movement.
       example: '1500'
+
+xiaomi_clean_zone_start:
+  description: Start the cleaning operation in the areas selected for the number of reps indicated.
+  fields:
+    entity_id:
+      description: Name of the vacuum entity.
+      example: 'vacuum.xiaomi_vacuum_cleaner'
+    zone:
+      description: Array of zones. Each zone is an array of 4 integer value.
+      example: '[[23510,25311,25110,26361]]'
+    reps:
+      description: Number of cleaning reps for each zone between 1 and 3.
+      example: '1'
+    reps_template:
+      description: Template that rapresents the number of cleaning reps for each zone between 1 and 3..
+      example: '{{states.input_number.vacuum_passes.state|int}}'
```

Comment:
I don't think template magic should be mixed in with functionality, but you should adjust your automations accordingly instead of this.

Reply:
It's the only way I have found to pass input_number values. Have you any other weay to suggest me?
Value to pass from Home Assistant: '{{states.input_number.vacuum_passes.state|int}}'

Reply:
The proper way to do this in my opinion is to simply pass the variable to the service call, maybe a better format for the payload would be:
```
entity_id: <id>
clean_zones: [<zone 1>, <zone 2>]
```
where zones are defined as dictionaries:
```
{
  'coordinates': [x1, x2, y1, y2],
  'repeats': 2
}
```
Then you can use regular automations to read the value from another entity and pass it to the service call, if I'm not mistaken?

Reply:
On the theory I agree with you, but Home assistant not resolve the "template formula" before pass data to .py.
The only way, I found, to pass the value of an input_number is to use the template magic and resolve it in .py.
If you help me to pass a value of input_number (like {{states.input_number.vacuum_passes.state|int}}) I'll so happy to remove template parameter.